### DISCUSSION TITLE
Google Trackers (Firebase Crashlytics) found in Github releases (Inside Job)

### DISCUSSION THREAD

**Discussion by ghost** [celzero/rethink-app#1568] [Q&A]
```diff
-Github release 41 - Inside Job, has Firebase Analytics and Crashlytics
- as well as release 42. No other compiled releases have these libs,
- on f-droid or google play. Is there an explanation as to why these are included?
```

**Comment by ignoramous** [comment]
Our build scripts (should) make sure to NOT include Firebase for GitHub & F-Droid flavours, but only for Play Store (which also contains other Google-related APIs that Rethink uses; for example to prompt user for *app updates*): 

https://github.com/celzero/rethink-app/blob/6fc4c28d02cb3b84066b7e0dc432096c4a494ad6/app/build.gradle#L294-L296

We've found Firebase Crashlytics in our own testing to be super good at capturing logs relating to crashes. We intend to use it to stablize the app (as Rethink crashing in the background and thus "leaking" connections is also a *major* problem). We did take a stab at this with our built-in "bug reporter", and we spent considerable time on, but it has proven to be a too much of a hassle with its own bugs. Users who do report bugs on GitHub often don't follow-up with logs or bug reports making rectifying those issues (despite a LOT of back and forth) without any progress towards a fix.

Rethink **does not** use Firebase "Analytics", just Crashlytics. More: #1455 

Tools like [Exodus](https://github.com/Exodus-Privacy/exodus-core) (which is also used by other Android app scanners like [Pithus](https://github.com/Pithus/bazaar)) incorrectly label apps with "Analytics": https://github.com/Exodus-Privacy/exodus-core/issues/97

On Play Store versions, users can *disable* Crashlytics auto-reporting crashes.

With *Loopback* mode (*Configure* -> *Network*), users can redirect Rethink's own traffic within Rethink (except "connectivity checks" if enabled) to monitor it just like any other app. In the present versions, we don't let users create rules on Rethink, but that's a temporary measure, as blocking Rethink itself may cause confusion.

**Reply by ignoramous** [reply]
I've added a note[^f] on GitHub releases[^g][^h] and the website[^i] about Crashlytics-related services sneaking through the build (it isn't clear why this happened when subsequent versions don't have it as the code that builds and excludes Crashlytics from GitHub and F-Droid flavours hasn't changed in any meaningful way).

[^f]: Quote: "It has been brought to our attention that this version contains Crashlytics (reports: [v055j](https://beta.pithus.org/report/778548e2c71f54a89c809b4dac288a08194955f4c72eb13ba6a55b66d92b6cb9), [v055k](https://beta.pithus.org/report/0a21c85ffafac3cdb4623ed0c762662ad7e946f6490114b69d2d17ec37c03bef)). If so, it has been included by mistake. More: https://github.com/celzero/rethink-app/discussions/1568"
[^g]: [v055j](https://github.com/celzero/rethink-app/releases/v0.5.5j)
[^h]: [v055k](https://github.com/celzero/rethink-app/releases/v0.5.5k)
[^i]: Quote: "Temporarily includes Crashlytics."

**Reply by ignoramous** [reply]
We have removed Crashlytics from *all* builds.
